skbio/metadata/_interval.py

```diff
@@ -0,0 +1,274 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from ._intersection import IntervalTree
+
+
+class Interval():
```

Comment:
For instance `boundaries` and `intervals`.

Perhaps I'm not on the same page.  I don't see the point of making this
subclass `collections.Mapping`.  Can you specify precisely what would be
some benefits of inheriting this?

On Wed, May 18, 2016 at 3:31 PM, Zech Xu <notifications@github.com> wrote:

> In skbio/metadata/_interval.py
> <https://github.com/biocore/scikit-bio/pull/1356#discussion_r63794475>:
>
> > @@ -0,0 +1,274 @@
> > +# ----------------------------------------------------------------------------
> > +# Copyright (c) 2013--, scikit-bio development team.
> > +#
> > +# Distributed under the terms of the Modified BSD License.
> > +#
> > +# The full license is in the file COPYING.txt, distributed with this software.
> > +# ----------------------------------------------------------------------------
> > +
> > +from ._intersection import IntervalTree
> > +
> > +
> > +class Interval():
>
> what name clashing?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/scikit-bio/pull/1356/files/6f54ec13b96ba8c8d97a6eef9fd37c5fa68cddf7#r63794475>
>